### PR TITLE
Enforce naming conventions for learning methods

### PR DESCRIPTION
**Pull request by jacobmontiel**
Changes proposed in this pull request:

* This PR addresses #138 
* Minor updates to MultiOutputLearner
  - Update documentation to explicitly indicate that this estimator is task-agnostic and supports both classification and regression depending on the type of base estimator.
  - Add base estimator type check
  - Add regression task test
* Add naming convention for new methods in CONTRIBUTING.md

---

Checklist

- [x] Code complies with PEP-8 and is consistent with the framework.
- [x] Code is properly documented.
- [x] Tests are included for new functionality or updated accordingly.
- [x] Travis CI build passes with no errors.
- [x] Test Coverage is maintained (threshold is -0.2%).
- [x] Files changed (update, add, delete) are in the PR's scope (no extra files are included).

### CODE REVIEW

File: src/skmultiflow/demos/_test_mtr.py [hunk 1]
```diff
@@ -1,20 +1,18 @@
-from skmultiflow.core.pipeline import Pipeline
 from skmultiflow.data import RegressionGenerator
 from skmultiflow.evaluation.evaluate_prequential import EvaluatePrequential
-from skmultiflow.trees.multi_target_regression_hoeffding_tree import \
-    MultiTargetRegressionHoeffdingTree
+from skmultiflow.trees.isoup_tree import iSOUPTreeRegressor
```
Comment (smastelini) [inline src/skmultiflow/demos/_test_mtr.py]: Would not be better to import `iSOUPTreeRegressor` directly from the `trees` submodule?

File: src/skmultiflow/trees/arf_hoeffding_tree.py [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
-from skmultiflow.trees.hoeffding_tree import HoeffdingTree, MAJORITY_CLASS, \
+from skmultiflow.trees.hoeffding_tree import HoeffdingTreeClassifier, MAJORITY_CLASS, \
     NAIVE_BAYES
 from skmultiflow.trees.nodes import RandomLearningNodeClassification
 from skmultiflow.trees.nodes import RandomLearningNodeNB
 from skmultiflow.trees.nodes import RandomLearningNodeNBAdaptive
 
 
-class ARFHoeffdingTree(HoeffdingTree):
+class ARFHoeffdingTree(HoeffdingTreeClassifier):
```
Comment (smastelini) [inline src/skmultiflow/trees/arf_hoeffding_tree.py]: Given that `ARFHoeffdingTree` is the base estimator of `AdaptiveRandomForestClassifier` (and `AdaptiveRandomForestRegressor` is coming on the way), should not its name carry the "classifier" sufix?

File: src/skmultiflow/trees/nodes/active_learning_node_for_regression.py [hunk 3]
```diff
@@ -30,7 +30,7 @@ def learn_from_instance(self, X, y, weight, ht):
             Instance target value.
         weight: float
             Instance weight.
-        ht: HoeffdingTree
+        ht: HoeffdingTreeClassifier
```
Comment (smastelini) [inline src/skmultiflow/trees/nodes/active_learning_node_for_regression.py]: Here, I believe that `HoeffdingTreeRegressor` would be the correct option.

File: src/skmultiflow/trees/nodes/active_learning_node_for_regression_multi_target.py [hunk 4]
```diff
@@ -30,7 +30,7 @@ def learn_from_instance(self, X, y, weight, ht):
             Targets values.
         weight: float
             Instance weight.
-        ht: HoeffdingTree
+        ht: HoeffdingTreeClassifier
```
Comment (smastelini) [inline src/skmultiflow/trees/nodes/active_learning_node_for_regression_multi_target.py]: The same as before. My fault here. I forgot to change the description when implementing the node classes.

File: tests/meta/test_multi_output_learner.py [hunk 5]
```diff
@@ -41,7 +44,7 @@ def test_multi_output_learner():
         classifier.partial_fit(X, y)
         cnt += 1
 
-    if not sklearn_version.startswith("0.21"):
+    if StrictVersion(sklearn_version) < StrictVersion("0.21"):
```
Comment (smastelini) [inline tests/meta/test_multi_output_learner.py]: Nice catch again!

File: src/skmultiflow/demos/_test_mtr.py (hunk 1 above)
Comment (jacobmontiel) [reply src/skmultiflow/demos/_test_mtr.py]: Agree. Updated in fdc3750b4e53c4c24cfd9b1d5127107543969262

File: src/skmultiflow/demos/_test_mtr.py [hunk 6]
```diff
@@ -1,20 +1,18 @@
...
 
 
 def demo(output_file=None):
-    """ _test_mtr_regression
+    """ Test iSOUP-Tree
 
-    This demo demonstrates how to evaluate a Multi-Target Regressor. The
-    employed dataset is 'scm1d', which is contained in the data folder.
+    This demo demonstrates how to evaluate a iSOUP-Tree multi-target regressor.
+    The employed dataset is 'scm1d', which is contained in the data folder.
```
[the remark this answers is not in the document]
Comment (jacobmontiel) [reply src/skmultiflow/demos/_test_mtr.py]: Fixed in fdc3750b4e53c4c24cfd9b1d5127107543969262

File: src/skmultiflow/demos/_test_mtr.py [hunk 7]
```diff
@@ -1,20 +1,18 @@
...
 
     Parameters
     ----------
     input_file: string
-        A string describind the path for the input dataset
+        A string describing the path for the input dataset
```
[the remark this answers is not in the document]
Comment (jacobmontiel) [reply src/skmultiflow/demos/_test_mtr.py]: Fixed in fdc3750b4e53c4c24cfd9b1d5127107543969262

File: src/skmultiflow/demos/_test_mtr.py [hunk 8]
```diff
@@ -25,10 +23,7 @@ def demo(output_file=None):
                                  n_targets=7)
     stream.prepare_for_use()
 
-    classifier = MultiTargetRegressionHoeffdingTree(leaf_prediction='adaptive')
-
-    # Setup the pipeline
-    pipe = Pipeline([('Classifier', classifier)])
+    classifier = iSOUPTreeRegressor(leaf_prediction='adaptive')
```
[the remark this answers is not in the document]
Comment (jacobmontiel) [reply src/skmultiflow/demos/_test_mtr.py]: Renamed in fdc3750b4e53c4c24cfd9b1d5127107543969262

File: src/skmultiflow/demos/_test_regression.py [hunk 9]
```diff
@@ -2,7 +2,7 @@
 from skmultiflow.core.pipeline import Pipeline
 from skmultiflow.evaluation.evaluate_prequential import EvaluatePrequential
 from skmultiflow.data.regression_generator import RegressionGenerator
-from skmultiflow.trees.regression_hoeffding_tree import RegressionHoeffdingTree
+from skmultiflow.trees.hoeffding_tree_regressor import HoeffdingTreeRegressor
```
[the remark this answers is not in the document]
Comment (jacobmontiel) [reply src/skmultiflow/demos/_test_regression.py]: Updated in fdc3750b4e53c4c24cfd9b1d5127107543969262

File: src/skmultiflow/trees/arf_hoeffding_tree.py (hunk 2 above)
Comment (jacobmontiel) [reply src/skmultiflow/trees/arf_hoeffding_tree.py]: Agree, renamed in a445c3716bba3a579434808008bad0b215109e6b

File: src/skmultiflow/trees/nodes/active_learning_node_for_regression.py (hunk 3 above)
Comment (jacobmontiel) [reply src/skmultiflow/trees/nodes/active_learning_node_for_regression.py]: Updated in 38da85c44c8c524c58704ec9e8718c554e313b0f

File: src/skmultiflow/trees/nodes/active_learning_node_for_regression_multi_target.py (hunk 4 above)
Comment (jacobmontiel) [reply src/skmultiflow/trees/nodes/active_learning_node_for_regression_multi_target.py]: Updated in 38da85c44c8c524c58704ec9e8718c554e313b0f

**Comment by jacobmontiel** [comment]
Thanks for the review @smastelini. I have addressed the action items. Regarding coverage, I have rely on

```python
# pragma: no cover
```

to skip the stubs for the old names.

File: src/skmultiflow/demos/_test_mtr.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (smastelini) [reply src/skmultiflow/demos/_test_mtr.py]: Thank you for the changes!

File: src/skmultiflow/demos/_test_mtr.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (smastelini) [reply src/skmultiflow/demos/_test_mtr.py]: Thank you for changing that!

**Comment by jacobmontiel** [comment]
Yes, all methods listed in #138 have been renamed.

**Comment by smastelini** [comment]
Im my opinion, this PR is ready to be merged. Thank you again @jacobmontiel for planning and implementing these changes.

**Pull request merged**